katdal/test/test_categorical.py

```diff
@@ -16,3 +16,19 @@ def test_dump_to_event_parsing():
     assert_array_equal(cleaned, [0, 2, 4, 6, 7], 'Dump->event parser failed')
     assert_array_equal(new_values, list('ACEGH'), 'Dump->event parser failed')
     assert_array_equal(new_events, [0, 1, 3, 5, 6], 'Dump->event parser failed')
+
+
+def test_categorical_sensor_creation():
+    timestamps = [-363.784, 2.467, 8.839, 8.867, 15.924, 48.925, 54.897, 88.982]
+    values = ['stop', 'slew', 'track', 'slew', 'track', 'slew', 'track', 'slew']
+    dump_period = 8.
+    dump_times = np.arange(4., 100., dump_period)
+    categ = sensor_to_categorical(timestamps, values, dump_times, dump_period,
+                                  greedy_values=('slew', 'stop'),
+                                  initial_value='slew')
+    assert_array_equal(categ.unique_values, ['slew', 'track'],
+                       'Sensor->categorical failed')
+    assert_array_equal(categ.events, [0, 2, 6, 7, 11, 12],
```

Comment:
It's probably just me, but is this correct? The slew at 48.925 starts during dump 5 (44-52), so greediness should make it take over dump 5, rather than only starting at dump 6.